pkgs/by-name/tr/treedome/package.nix

```diff
@@ -80,47 +75,33 @@ rustPlatform.buildRustPackage {
   # Also modify tauri.conf.json so that it expects the resources at the new location
   postPatch = ''
     substituteInPlace ./tauri.conf.json \
-      --replace '"distDir": "../dist",' '"distDir": "dist",' \
-      --replace '"beforeBuildCommand": "yarn run build",' '"beforeBuildCommand": "",'
+      --replace-fail '"frontendDist": "../dist",' '"frontendDist": "dist",' \
+      --replace-fail '"beforeBuildCommand": "yarn run build",' '"beforeBuildCommand": "",'
```

Comment:
```suggestion
```
I'd also recommend removing this patch and the split derivation for the front end. `cargo-tauri` and the hook should be able to handle the build itself using this information, as long as you move the `offlineCache` to this derivation and add `yarnConfigHook` to `nativeBuildInputs`

You can see an example of this with npm in the [hook's documentation](https://github.com/NixOS/nixpkgs/blob/7f86ed92ccae28a4fe69d8a62b6088766f3d9022/doc/hooks/tauri.section.md#example-code-snippet-tauri-hook-example-code-snippet)

Reply:
I don't get this one, what do you mean by splitting? Do you mean merging, as in making the derivation of frontend and backend into one?

Also about cargo-tauri, maybe I will try it without this patch, but will it really work? Does cargo-tauri.hook know exactly where the build is even when the path is wrong in tauri.conf.json?

Reply:
> I don't get this one, what do you mean by splitting? Do you mean merging, as in making the derivation of frontend and backend into one?

Yes

> Also about cargo-tauri, maybe I will try it without this patch, but will it really work?

It has with every Tauri package i've seen

> Does cargo-tauri.hook know exactly where the build is even when the path is wrong in tauri.conf.json?

And the path won't be wrong if we let Tauri handle the big correctly (unless upstream has majorly misconfigured something)

Reply:
I've made it into a single file now